client/profile-wizard/index.js

```diff
@@ -203,16 +201,63 @@ class ProfileWizard extends Component {
 	}
 
 	completeProfiler() {
-		const { notes, updateNote, updateProfileItems } = this.props;
-		updateProfileItems( { completed: true } );
+		const {
+			activePlugins,
+			getJetpackConnectUrl,
+			getPluginsError,
+			isJetpackConnected,
+			notes,
+			updateNote,
+			updateProfileItems,
+		} = this.props;
 		recordEvent( 'storeprofiler_complete' );
+		const shouldConnectJetpack =
+			activePlugins.includes( 'jetpack' ) && ! isJetpackConnected;
 
 		const profilerNote = notes.find(
 			( note ) => note.name === 'wc-admin-onboarding-profiler-reminder'
 		);
 		if ( profilerNote ) {
 			updateNote( profilerNote.id, { status: 'actioned' } );
 		}
+
+		const promises = [
+			updateProfileItems( { completed: true } ).then( () => {
+				if ( shouldConnectJetpack ) {
+					document.body.classList.add(
+						'woocommerce-admin-is-loading'
+					);
+				}
+			} ),
+		];
+
+		if ( shouldConnectJetpack ) {
+			promises.push(
+				getJetpackConnectUrl( {
+					redirect_url: getAdminLink( 'admin.php?page=wc-admin' ),
+				} )
+			);
+		}
+
+		Promise.all( promises ).then( ( responses ) => {
+			if ( ! shouldConnectJetpack ) {
+				return;
+			}
+
+			const jetpackConnectUrl = responses[ 1 ];
```

Comment:
This feels a little brittle, could you add a comment explaining why it's index 1? That way if a new promise is inserted at least there's an explanation why the jetpack redirect has stopped working.

Reply:
Refactored in https://github.com/woocommerce/woocommerce-admin/pull/4717/commits/fc6a9c4900831c872eacc61c26538b31db1fa174 so we can handle redirection regardless of whether or not a promise exists.